src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -104,11 +106,14 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             final Marshaller marshaller = jaxbContext.createMarshaller();
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
             marshaller.marshal(toSave, fileWriter);
+            isWritable( path );
 
         } catch (IOException ioe) {
             throw new StorageOperationException("Error writing to file: " + path);
         } catch (JAXBException jaxbe) {
             throw new StorageOperationException("Error converting address book into storage format");
+        }  catch (SecurityException e) {
+            throw new StorageOperationException("This file at "+ path +" may not be writable. Please try another file. ");
```

Comment:
You can use the exception directly to handle it instead of checking if the file `isWritable()` above.